services/QuillLMS/client/app/bundles/Teacher/components/premium/school_and_district_premium_modal.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+
+import SchoolSelector, { NOT_LISTED, } from '../shared/school_selector'
+import { requestPost, requestPut, } from '../../../../modules/request';
+import { closeIcon, DropdownInput, Spinner, } from '../../../Shared/index'
+
+export const SCHOOL_PURCHASE_STAGE_TWO = 'school_purchase_stage_two'
+
+const SchoolAndDistrictPremiumModal = ({ stripeSchoolPlan, associatedSchools, userIsSignedIn, startAtStageTwo, customerEmail, closeModal, handleNotListedSelection, }) => {
+  const [stage, setStage] = React.useState(startAtStageTwo ? 2 : 1)
+  const [selectedSchool, setSelectedSchool] = React.useState(null)
+  const [loading, setLoading] = React.useState(false)
+
+  React.useEffect(handleStageTwo, [])
```

Comment:
Is this needed?

Reply:
Yes - this handles when we are stage two as of initial load (because the user was sent to login and then redirected back here with the query param). The following effect will only run when the `stage` has been changed. Unfortunately there isn't a more succinct way to say "run this hook both at initial load and when the stage changes".